### PR TITLE
handle the slash '/' command to open the insert plugin sidebar #982

### PR DESCRIPTION
**Pull request by karthick3018**
To Open the Cell Plugins on pressing the slash "/" key
 
## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

- [x] Lint and unit tests pass locally with my changes
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by karthick3018** [comment]
fixes #982

File: packages/editor/src/core/components/Cell/InsertNew.tsx [hunk 1]
```diff
@@ -31,6 +31,25 @@ const InsertNew: React.FC<{ parentCellId?: string }> = ({ parentCellId }) => {
     },
   });
 
+  React.useEffect(() => {
+    window?.addEventListener('keydown', handleKeyDown);
+    return () => {
+      window?.removeEventListener('keydown', handleKeyDown);
+    };
+  }, []);
```
Comment (macrozone) [inline packages/editor/src/core/components/Cell/InsertNew.tsx:39]: handleKeyDown is used before define, you should move the declaration up. 
You can declare it in `useEffect` directly. Make sure to add `handleInsertMode` to the deps array then.

File: packages/editor/src/core/components/Cell/InsertNew.tsx [hunk 2]
```diff
@@ -31,6 +31,25 @@ const InsertNew: React.FC<{ parentCellId?: string }> = ({ parentCellId }) => {
...
+
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e?.key === '/') {
```
Comment (macrozone) [inline packages/editor/src/core/components/Cell/InsertNew.tsx:42]: can we make this configurable? (could also be done later)

**Review by macrozone — CHANGES_REQUESTED** [review]
The effect was added in `InsertNew`. This component is mounted multiple times (e.g. in cell plugins that take other cells). Additionaly this component has a reference node id (of the parent cell), which is responsible to show the plugins that can be added to that parent cell.

The key listener effect should therefore be moved in a component that is only mounted once

File: packages/editor/src/core/components/Cell/InsertNew.tsx (hunk 1 above)
Comment (karthick3018) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:39]: can you specify which file ? for better understanding

File: packages/editor/src/core/components/Cell/InsertNew.tsx (hunk 2 above)
Comment (karthick3018) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:42]: I didn't get this point throw an example like how you're expecting

Comment (macrozone) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:42]: i am open for a suggestion, but we can omit it for the time being

File: packages/editor/src/core/components/Cell/InsertNew.tsx (hunk 1 above)
Comment (macrozone) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:39]: i would move it somewhere in EditableEditor

Comment (karthick3018) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:39]: packages/editor/src/editor/EditableEditor.tsx -> this file doesn't have the redux connect to handle the useSetInsertMode hook . packages/editor/src/core/components/Editable/index.tsx -> shall we move the logic to this file?

Comment (macrozone) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:39]: best wrap it in a component (`<GlobalHotkeys />`) and add that to the `<EditableEditor />` inside the `<Provider />`

then this `GlobalHotkeys` just returns null, but has a `useEffect` hook.

Comment (karthick3018) [reply packages/editor/src/core/components/Cell/InsertNew.tsx:39]: @macrozone updated !

File: packages/editor/src/core/components/Cell/InsertNew.tsx [hunk 3]
```diff
@@ -31,6 +31,25 @@ const InsertNew: React.FC<{ parentCellId?: string }> = ({ parentCellId }) => {
...
+      handleInsertMode(e);
+    }
+  };
+
+  const handleInsertMode = (e) => {
+    e.stopPropagation();
+    setReferenceNodeId(parentCellId);
+    setInsertMode();
+  };
+
```
Comment (macrozone) [inline packages/editor/src/core/components/Cell/InsertNew.tsx:52]: you can now undo all these changes here

**Review by macrozone — CHANGES_REQUESTED** [review]

File: packages/editor/src/core/components/Cell/InsertNew.tsx (hunk 3 above)
Comment (karthick3018) [reply packages/editor/src/core/components/Cell/InsertNew.tsx]: @macrozone all changes done !!

**Comment by karthick3018** [comment]
thanks @macrozone

**Comment by macrozone** [comment]
@karthick3018 sorry, the feature was not release as the linting failed. Sorry i did not check that and fix that on master.

I will probably need to introduce precommit hooks to mitigate that as sometimes devs have no linter running locally

**Comment by karthick3018** [comment]
@macrozone  do I need to fix anything and commit to solve those linting ?

**Comment by macrozone** [comment]
> @macrozone do I need to fix anything and commit to solve those linting ?

no, just did it. (i just ran `yarn lint --fix`)

**Comment by karthick3018** [comment]
@macrozone relaxing comment!!! thanks

**Comment by macrozone** [comment]
@karthick3018  ;-)

travis has some problems currently as it seems, might take some more minutes to build (usually it builds in under 5min)

**Comment by karthick3018** [comment]
@macrozone hope it will resolve soon , event listener PR shouldn't cause this much trouble !!lolz

**Pull request merged**